### ISSUE TITLE
Fix DeepSeekV3 demo outputs after tt-metal submodule uplift

### ISSUE THREAD

**Issue by esmalTT** [tenstorrent/tt-metal#48494]
## Summary

After the tt-metal submodule uplift to v0.73 (branch `uplift-run-2-bf7-20260624`, submodule `98f1b636` → `e9ff853ff`), the **DeepSeek-V3 demo produces bad outputs**: generation starts on-topic for ~10–12 tokens and then **switches to a completely different, unrelated response**.

This is an **accuracy** regression, not a hang or perf issue — TTFT and decode throughput are normal.

## Bad output

The problem is that after the first ~10–12 decode tokens the model **stops following the input prompt and produces an unrelated response**, as if it has lost the prompt context partway through decode.

Prompt (`DeepSeek-R1-0528-dequantized`, greedy decode):
> "Solve this step by step: Design a cache for sharded LLM weights across multiple hosts with local NVMe and shared NFS. Minimize startup latency, avoid duplicate reads, support cache invalidation, and explain tradeoffs between per-host caches, content-addressable storage, and pre-sharded artifacts."

Model output:
```
<think>
We are designing a cache for sharded# 1.0.0 (2023-05-05)
...
```
- The first ~10–12 tokens are on-topic ("`We are designing a cache for sharded`").
- It then abruptly switches to unrelated-but-coherent content (e.g. a "`# 1.0.0 (2023-05-05)`" changelog-style fragment) and keeps generating reasonable English/code that has nothing to do with the prompt.
- Generation ends at EOS (token id 1) on decode step 102.

For reference, **Kimi-K2 — which shares most of its implementation with DeepSeek-V3 (MLA attention + DeepSeek-style MoE) — generates 2048 fully-coherent, on-topic tokens on the same commit.** So the regression is specific to a DeepSeek-V3 code path not shared with Kimi.

## Reproduce on Aisle A SC16

Multi-host superpod run:
```
TT_METAL_ALLOCATOR_MODE_HYBRID=1 TT_METAL_SLOW_DISPATCH_MODE=1 tt-run \
  --tcp-interface ens5f0np0 \
  --mpi-args "--map-by rankfile:file=rev_c_rank_file_superpod --bind-to hwt:overload-allowed --host ${HOSTSP} --tag-output" \
  --rank-binding rev_c_rank_binding_superpod.yaml \
  python -m blaze.models.cli \
  --model deepseek-ai/DeepSeek-V3 \
  --weights real --cache-path /home/$USER/cache \
  --model-path /mnt/models/deepseek-ai/DeepSeek-R1-0528-dequantized \
  --max-new-tokens 2048 \
  --prompt "Solve this step by step: ..."
```

## Status

Investigation in progress — see the comments for the full breakdown. Key result so far: **every single-host-testable component is bit-clean with real weights** (MLA attention PCC ≈ 0.9999 up to pos 8191; MoE PCC ≈ 0.999996; full decoder layer ≈ 0.9996), and the `eltwise_mul_scalar` revert, the SFPU exp/sigmoid rewrite, and the slots/metadata/RoPE paths were all ruled out on hardware. The "loses prompt context mid-decode" signature plus the clean isolated numerics point at a problem in the **multi-stage superpod path** (inter-stage `d2d_exchange` pipeline sockets and/or multi-pod CCL — i.e. how prompt/KV context is carried across pipeline stages), which currently has no CI coverage for DeepSeek-V3.

**Comment by esmalTT** [comment]
## Investigation summary — DeepSeek-V3 demo collapse on the uplift branch

### Hardware A/B results (BH Galaxy, real weights, `--weights state_dict`)
Single-galaxy isolated tests, real DeepSeek-V3 weights:

| Test | Metric | PCC |
|------|--------|-----|
| `test_backed_mla[deepseek_v3]` | MLA attention vs golden, pos 1023 | 0.99995 |
| `test_backed_mla[deepseek_v3]` | MLA attention vs golden, pos 4096 | 0.99998 |
| `test_backed_mla[deepseek_v3]` | MLA attention vs golden, pos 8191 | 0.99999 |
| `test_sparse_layer_backed[deepseek_v3, pos4096]` | MLA golden PCC | 0.99959 |
| `test_sparse_layer_backed[deepseek_v3, pos4096]` | **MoE PCC** | **0.9999960** |
| `test_sparse_layer_backed[deepseek_v3, pos4096]` | end-to-end decoder layer PCC | 0.99958 |

Every isolated single-host DeepSeek-V3 component is **bit-clean with real weights**, including at long KV length.

> Note: `test_routed_expert_blaze[deepseek_v3]` reports PCC 0.9798, but that test uses **synthetic random expert weights** (adversarial for numerics). With real weights the same MoE path is 0.999996 — the 0.9798 is a test artifact, not a regression.

### Hypotheses tested and **refuted**
- **`eltwise_mul_scalar` #1597 workaround revert (`537dd21a0`)** — re-ran `test_routed_expert` with and without the revert: PCC **bit-identical** (`0.9798207226699006` both ways). The `(icb0,icb0)` vs `DEFAULT_TENSOR_SHAPE` init forms are numerically equivalent on this submodule pin; #44602 does **not** reproduce the old PCC-0.726 break here.
- **SFPU `exp`/`sigmoid` rewrite (#45642 / #45594) via attention softmax** — MLA softmax PCC is 0.9999 even at pos 8191 and does not degrade with KV length. DeepSeek-V3 attention runs `DST_ACCUM_MODE=false` (bf16/approx exp), not the rewritten fp32-accurate exp, so the upstream "softmax 0.99→0.82" effect does not reach this path.
- **Slots / metadata (`position_id`,`slot_id`) / RoPE** — metadata byte layout intact (64 B) and the `CclBroadcast` #1673 multi-input rewrite preserves metadata contiguity (hard-checked at compose time); KV/slot addressing is symmetric with no boundary near the collapse point; RoPE cos/sin is precomputed **host-side in torch fp32** (device only multiplies a precomputed bf16 table), so the SFPU trig rewrite cannot reach it. All three are identical to Kimi's path.
- **Per-layer / depth compounding of numerical error** — single decoder layer is too clean (0.9996) to explain the collapse.

**Comment by esmalTT** [comment]
Stale - this was resolved a while ago
